cmd/destroy/remote.go

```diff
@@ -284,161 +136,26 @@ func (rd *remoteDestroyCommand) destroy(ctx context.Context, opts *Options) erro
-		GitBranchArgName:       constants.OktetoGitBranchEnvVar,
-		InvalidateCacheArgName: constants.OktetoInvalidateCacheEnvVar,
-		DestroyFlags:           strings.Join(getDestroyFlags(opts), " "),
-	}
-
-	dockerfile, err := rd.fs.Create(filepath.Join(tempDir, dockerfileTemporalNane))
-	if err != nil {
-		return "", err
-	}
-
-	if err = remote.CreateDockerignoreFileWithFilesystem(cwd, tempDir, filesystem.CleanManifestPath(opts.ManifestPathFlag), rd.fs); err != nil {
-		return "", err
-	}
-
-	if err := tmpl.Execute(dockerfile, dockerfileSyntax); err != nil {
-		return "", err
-	}
-	return dockerfile.Name(), nil
-
-}
-
-func getExtraHosts(registryURL, subdomain, ip string, metadata types.ClusterMetadata) []types.HostMap {
-	extraHosts := []types.HostMap{
-		{Hostname: registryURL, IP: ip},
-		{Hostname: fmt.Sprintf("kubernetes.%s", subdomain), IP: ip},
-	}
-
-	if metadata.BuildKitInternalIP != "" {
-		extraHosts = append(extraHosts, types.HostMap{Hostname: fmt.Sprintf("buildkit.%s", subdomain), IP: metadata.BuildKitInternalIP})
-	}
-
-	if metadata.PublicDomain != "" {
-		extraHosts = append(extraHosts, types.HostMap{Hostname: metadata.PublicDomain, IP: ip})
-	}
-
-	return extraHosts
-}
-
-func getDestroyFlags(opts *Options) []string {
-	var deployFlags []string
-
-	if opts.Name != "" {
-		deployFlags = append(deployFlags, fmt.Sprintf("--name \"%s\"", opts.Name))
-	}
-
-	if opts.Namespace != "" {
-		deployFlags = append(deployFlags, fmt.Sprintf("--namespace %s", opts.Namespace))
-	}
-
-	if opts.ManifestPathFlag != "" {
-		deployFlags = append(deployFlags, fmt.Sprintf("--file %s", filesystem.CleanManifestPath(opts.ManifestPathFlag)))
-	}
-
-	if opts.DestroyVolumes {
-		deployFlags = append(deployFlags, "--volumes")
+func getCommandFlags(opts *Options) ([]string, error) {
+	commandFlags := []string{
+		fmt.Sprintf("--name \"%s\"", opts.Name),
 	}
 
 	if opts.ForceDestroy {
-		deployFlags = append(deployFlags, "--force-destroy")
+		commandFlags = append(commandFlags, "--force-destroy")
 	}
 
-	return deployFlags
-}
-
-func getOktetoCLIVersion(versionString string) string {
-	var version string
-	if match, err := regexp.MatchString(`\d+\.\d+\.\d+`, versionString); match {
-		version = fmt.Sprintf(constants.OktetoCLIImageForRemoteTemplate, versionString)
-	} else {
+	if len(opts.Variables) > 0 {
+		var varsToAddForDeploy []string
+		variables, err := env.Parse(opts.Variables)
 		if err != nil {
-			oktetoLog.Infof("invalid okteto CLI version %s: %s. Using latest", versionString, err)
-		} else {
-			oktetoLog.Infof("invalid okteto CLI version %s. Using latest", versionString)
+			return nil, err
 		}
-		remoteOktetoImage := os.Getenv(constants.OktetoDeployRemoteImage)
-		if remoteOktetoImage != "" {
-			version = remoteOktetoImage
-		} else {
-			version = fmt.Sprintf(constants.OktetoCLIImageForRemoteTemplate, "latest")
+		for _, v := range variables {
+			varsToAddForDeploy = append(varsToAddForDeploy, fmt.Sprintf("--var %s=%q", v.Name, v.Value))
```

Comment:
I'm adding the `var` flag to the destroy operation as it is the way we pass the variables to the `remote-run destroy` command